File: w4_array_16/tory/three_sum_closest.py
from typing import List


class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        n = len(nums)
        res = float("inf")
        for i in range(n):
            if i > 0 and nums[i] == nums[i - 1]:
                continue
            left = i + 1
            right = n - 1
            while left < right:
                cur = nums[i] + nums[left] + nums[right]
                if cur == target: return target
                if abs(res - target) > abs(cur - target):
                    res = cur
                if cur > target:
                    right -= 1
                elif cur < target:
                    left += 1
        return res

File: w4_array_16/lijialu/threeSumClosest.go
package main

import (
	"fmt"
	"math"
)

func main() {
	nums := []int{-1, 2, 1,-4}
	fmt.Println(threeSumClosest(nums, 1))
}

func threeSumClosest(nums []int, target int) int {
	var sum int
	var diff int
	var res int
	minDiff := math.MaxInt16

	length := len(nums) -1
	i := 0


	for ; i <= length; i++ {
		for j:=0; j <= length; j++ {
			if i == j {
				continue
			}
			for k:=0; k <= length; k++ {
				if i == k || j ==k {
					continue
				}

				sum = nums[i] + nums[j] + nums[k]
				fmt.Println(i,j,k,nums[i],nums[j],nums[k])
				diff = int(math.Abs(float64(sum-target)))
				if diff < minDiff {
					minDiff = diff
					res = sum
				}
			}
		}
	}

	return res
}
